### DISCUSSION TITLE
Unable to federate with my own instance from misskey.io. Figuring out the best way to troubleshoot it.

### DISCUSSION THREAD

**Discussion by jojobii-arks** [misskey-dev/misskey#9194] [Q&A]
Pulled from #9180. It's more of a troubleshooting issue, rather than an actual bug.

---

## 💡 Summary

I'm unable to view accounts and posts from my instance on `misskey.io`, despite not being banned. I've tried to view my account via UI and API Console (`/users/show`).

I can view `misskey.io` accounts and posts from my instance, but I am unable to follow accounts. The follow button gets stuck at `"Processing..."` after clicking, even after waiting a few days. 

My instance name is `mk.arks.cafe`.

I've checked with other instances running the same version of Misskey, `v12.119.0`, but they all seem to work fine. It's only the connection from `misskey.io` to `mk.arks.cafe` where the problem occurs; it's tough to troubleshoot...

## 🥰 Expected Behavior

Proper interaction with `mk.arks.cafe` from `misskey.io`, including being able to view posts and users from `mk.arks.cafe`.

## 📝 Steps to Reproduce

1. While signed in on `mk.arks.cafe`, go to https://mk.arks.cafe/@syuilo@misskey.io and attempt to follow. Observe "Processing..."

<img width="786" alt="CleanShot 2022-11-15 at 15 27 12@2x" src="https://user-images.githubusercontent.com/78003700/202046245-b7a244a6-cea6-4bfc-955b-8978441a626f.png">

2. While signed in on `misskey.io`, attempt to view user `@jojobii@mk.arks.cafe`, either by URL or api endpoint (`users/show).

<img width="647" alt="CleanShot 2022-11-15 at 15 21 20@2x" src="https://user-images.githubusercontent.com/78003700/202045542-1b65735e-73ae-46bd-bc67-7b1e69504bc5.png">

3. View `misskey.io` federation and search for `mk.arks.cafe`

<img width="1435" alt="CleanShot 2022-11-15 at 15 12 54@2x" src="https://user-images.githubusercontent.com/78003700/202044531-5bafca80-ea19-4b13-af51-ae23c60f86a8.png">

4. From another instance, such as `miruku.cafe`, view account from `mk.arks.cafe` (https://miruku.cafe/@jojobii@mk.arks.cafe)

<img width="824" alt="CleanShot 2022-11-15 at 15 33 25@2x" src="https://user-images.githubusercontent.com/78003700/202046985-b9c140f1-ac10-4eb1-8f4e-b494ecc62e6f.png">

#### Server @ `mk.arks.cafe`

Hosting: DigitalOcean Droplet
Misskey version: v12.119.0
Node: v18.12.1
Postgres: v13.9
Redis: 7.0.5
Vue: 3.2.39

Your OS: MacOS 13
Your browser: Chrome

---

Any help or tips regarding troubleshooting and whatnot would be lovely! 🙇

**Comment by ThatOneCalculator** [comment]
This is a bit of a shot in the dark, but maybe try enabling `signToActivityPubGet` in your `.config/default.yml`?

**Reply by jojobii-arks** [reply]
Yep; set the value and rebuilt, no dice :/

Currently trying to `grep` my way through log files to see if I can find where things went wrong, but to no avail.

**Comment by belatedly** [comment]
I am having a similar issue, I had a domain name set up as a Plume instance for awhile and had followed my account on there on my Misskey instance. I thought deleting the user and instance from the database would refresh the federation when I moved it to an Akkoma instance, but even though the user and instance were found as new, Misskey isn't receiving any activities from the Akkoma instance but it is sending (I was able to submit a follower request to Akkoma, but when it was accepted, Misskey didn't read this). Wondering if our issues have a similar solution.

**Comment by jojobii-arks** [comment]
Hi, just coming back here to document the fact that it seemingly works now? I'm unsure what has changed besides enabling `signToActivityPubGet` and waiting a whole month, but maybe the wait is necessary as it needs to propogate? 

![image](https://user-images.githubusercontent.com/78003700/208719896-713979f0-2c7c-4277-8e62-8f484d424d79.png)
